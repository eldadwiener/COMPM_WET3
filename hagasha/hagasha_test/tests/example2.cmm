int power (c: int, m:int)
{
    d:int;
    if m==0 then return 1;
    if m==1 then return c;
    d=power(c,m-1);
    return d*c;
}

int power (c: int, m:int);

void main()
{
	write("\nMoodle Test 2 (recursion)\n");
    a:int;
    b:int;
    n:int;

    read(a);
    read(n);
    b = power (a,n);
    write("should be a^n : "); write(b); write("\n");
}
