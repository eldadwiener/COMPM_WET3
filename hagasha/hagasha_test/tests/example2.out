
Moodle Test 2 (recursion)
Input integer?:Input integer?:should be a^n : 1296 
Reached Halt.
